docs/src/details/main-modules/images/sources.rst

```diff
@@ -0,0 +1,125 @@
+.. include:: /include/external_links.txt
+
+.. _image_sources:
+
+=============
+Image Sources
+=============
+
+The images module supports images in any of 3 forms:
+
+- **variables** (i.e. image data in ROM or RAM):  a :cpp:type:`lv_image_dsc_t` object
+  plus its data
+- **files** (strings beginning with a printable ASCII character are interpreted as a
+  file-system path to an image file)
+- **strings** containing Unicode characters
+
+.. container:: tighter-table-3
+
+    +-----------------------+--------------+------------+
+    | Form                  | Compressed?  | Location   |
+    +=======================+==============+============+
+    | variable (C struct)   | No           | ROM        |
+    +-----------------------+--------------+------------+
+    | variable (C struct)   | No           | RAM        |
+    +-----------------------+--------------+------------+
+    | file                  | No           | .bmp file  |
+    +-----------------------+--------------+------------+
+    | file                  | Yes          | .png file  |
+    +-----------------------+--------------+------------+
+    | file                  | Yes          | .jpg file  |
+    +-----------------------+--------------+------------+
+    | file                  | Yes          | .gif file  |
+    +-----------------------+--------------+------------+
+    | buffer read from file | File content | RAM        |
+    +-----------------------+--------------+------------+
+    | Unicode string        | No           | ROM or RAM |
+    +-----------------------+--------------+------------+
+
+
+.. _image_sources_variables:
+
+Variables
+*********
+
+Images stored as variables are comprised of an :cpp:struct:`lv_image_dsc_t` struct
+plus its data.  The struct contains the following fields:
+
+:header:
+
+    :magic:       Magic number (must be LV_IMAGE_HEADER_MAGIC)
+    :cf:          Color format. See :ref:`images_color_formats`.
+    :flags:       Image flags (any of the ``LV_IMAGE_FLAGS_...`` enumeration values).
+    :w:           Width in pixels (<= 2048)
+    :h:           Height in pixels (<= 2048)
+    :stride:      Number of bytes between the beginning of one row of pixels and the beginning of the next row
+    :reserved_2:  reserved for future use
+
+:data_size:  length of ``data`` in bytes
+:data:       pointer to a byte array where the data of the image is stored (pixels
+             and sometimes indexed color palettes [for indexed color formats])
+
+These are usually stored within a project as a C file generated by one of these two
+LVGL image converters:
+
+- `LVGL Online Image Converter`_
```

Comment:
It might be worth adding a new heading here, e.g. "Converting Images to C array"